examples/08_interpolation_join.py

```diff
@@ -0,0 +1,173 @@
+"""
+Interpolation join: infer missing rows when joining two tables
+==============================================================
+
+We illustrate the ``InterpolationJoin``, which is a type of join where values from the second table are inferred with machine-learning, rather than looked up in the table.
+It is useful when exact matches are not available but we have rows that are close enough to make an educated guess -- in this sense it is a generalization of a ``fuzzy_join``.
+
+The ``InterpolationJoin`` is therefore a transformer that adds the outputs of one or more machine-learning models as new columns to the table it operates on.
+
+In this example we want our transformer to add weather data (temperature, rain, etc.) to the table it operates on.
+We have a table containing information about commercial flights, and we want to add information about the weather at the time and place where each flight took off.
+This could be useful to predict delays -- flights are often delayed by bad weather.
+We have a table of weather data containing, at many weather stations, measurements such as temperature, rain and snow at many time points.
+Unfortunately, our weather stations are not inside the airports, and the measurements are not timed according to the flight schedule.
+Therefore, a simple equi-join would not yield any matching pair of rows from our two tables.
+Instead, we use the ``InterpolationJoin`` to *infer* the temperature at the airport at take-off time.
+We train supervised machine-learning models using the weather table, then query them with the times and locations in the flights table.
+
+"""
+
+######################################################################
+# Load weather data
+# -----------------
+# We join the table containing the measurements to the table that contains the weather stations’ latitude and longitude.
+# We subsample these large tables for the example to run faster.
+
+from skrub.datasets import fetch_figshare
+import pandas as pd
+
+weather = fetch_figshare("41771457").X
+weather = weather.sample(100_000, random_state=0, ignore_index=True)
```

Comment:
I used `sample` because that is what is done in the "spatial join" [example](https://skrub-data.org/stable/auto_examples/07_multiple_key_join.html#sphx-glr-auto-examples-07-multiple-key-join-py). I think it's a good idea to take a random sample rather than the first rows in case the dataset is sorted by date or weather station. But I can check whether that is the case